pkg/kubernetes/deploy.js

```diff
@@ -73,6 +80,220 @@ define([
+    }
+
+
+    function install_nulecule() {
+        var promise = validate()
+            .fail(function(exs) {
+                $("#deploy-app-dialog").dialog("failure", exs);
+            })
+            .done(function(fields) {
+
+                var promise1 = nulecule_client.create_tmp()
+                    .done(function(tmp){
+                        nulecule_client.kill_atomicapp();
+                        install_dir = tmp.trim();
+
+                        var promise2 = nulecule_client.installrun("install", install_dir, fields.nulecule_image)
+                            .done(function() {
+
+                                var promise3 = nulecule_client.get_statuslist()
+                                    .done(function(ans_json){
+                                        answerfile = nulecule_client.loadAnswersfile(ans_json);
+
+                                        //Show App Params in the dialog
+                                        var template = $('#deploy-app-appentity-template').html();
+                                        Mustache.parse(template);
+                                        var text = Mustache.render(template, $.extend({
+                                            "apps": convertToKV(answerfile)
+                                        }));
+                                        $('#deploy-app-dialog').find('.cockpit-form-table').append(text);
+                                        
+                                        run_stage = true;
+                                        return;
+
+                                    })
+                                    .fail(function(ex, response) {
+                                        /* Display the error appropriately in the dialog */
+                                        $("#deploy-app-dialog").dialog("failure", ex);
+                                    });
+                            })
+                            .fail(function(ex, response) {                               
+                                /* Display the error appropriately in the dialog */
+                                $("#deploy-app-dialog").dialog("failure", ex);
+                            });
+
+
+                        /* Display a spinner while install is happening */
+                        $("#deploy-app-dialog").dialog("wait", promise2);
+                    })
+                    .fail(function(ex, response) {
+                        /* Display the error appropriately in the dialog */
+                        $("#deploy-app-dialog").dialog("failure", ex);
+                    });
+
+                    /* Display a spinner while tmp folder is happening */
+                    $("#deploy-app-dialog").dialog("wait", promise1);
+
+            });
+
+        /* Display a spinner while vaidation is happening */
+        $("#deploy-app-dialog").dialog("wait", promise);
+    }
+
+
+    function validate_app_fields() {
+        var dfd = $.Deferred();
```

Comment:
Since this function does not perform any asynchronous operations, it's not necessary that it returns a deferred promise.